components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/dao/impl/CORSOriginDAOImpl.java

```diff
@@ -0,0 +1,210 @@
+                CORSOrigin corsOrigin = new CORSOrigin();
+                corsOrigin.setId(resultSet.getString("ID"));
+                corsOrigin.setOrigin(resultSet.getString("ORIGIN"));
+                corsOrigins.add(corsOrigin);
+            }
+            return corsOrigins;
+        } catch (SQLException e) {
+            throw handleServerException(ERROR_CODE_CORS_RETRIEVE, e, tenantDomain);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, resultSet, preparedStatement);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<CORSOrigin> getCORSOriginsByApplicationId(String tenantDomain, String applicationId)
+            throws CORSManagementServiceServerException {
+
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        try {
+            preparedStatement = connection.prepareStatement(SQLConstants.GET_CORS_ORIGINS_BY_APPLICATION_ID);
+            preparedStatement.setString(1, tenantDomain);
+            preparedStatement.setString(2, applicationId);
+            resultSet = preparedStatement.executeQuery();
+
+            List<CORSOrigin> corsOrigins = new ArrayList<>();
+            while (resultSet.next()) {
+                CORSOrigin corsOrigin = new CORSOrigin();
+                corsOrigin.setId(resultSet.getString("ID"));
+                corsOrigin.setOrigin(resultSet.getString("ORIGIN"));
+                corsOrigins.add(corsOrigin);
+            }
+            return corsOrigins;
+        } catch (SQLException e) {
+            throw handleServerException(ERROR_CODE_CORS_RETRIEVE, e, tenantDomain);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, resultSet, preparedStatement);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addCORSOrigins(String tenantDomain, List<CORSOrigin> corsOrigins)
+            throws CORSManagementServiceServerException {
+
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        Connection retrieveConnection = IdentityDatabaseUtil.getDBConnection(false);
+        Connection commitConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try {
+            for (CORSOrigin corsOrigin : corsOrigins) {
+                preparedStatement = retrieveConnection.prepareStatement(SQLConstants.GET_TENANT_ID_BY_DOMAIN);
+                preparedStatement.setString(1, tenantDomain);
+                resultSet = preparedStatement.executeQuery();
+
+                String tenantId = null;
+                while (resultSet.next()) {
+                    tenantId = resultSet.getString("ID");
+                }
+
+                if (StringUtils.isBlank(tenantId)) {
+                    throw handleServerException(ERROR_CODE_CORS_ADD, tenantDomain);
+                }
```

Comment:
You can use the available util to resolve the tenantID. `IdentityTenantUtil.getTenantId(tenantDomain)`